src/PersonObjects/Player/PlayerObjectGeneralMethods.tsx

```diff
@@ -1854,7 +1854,7 @@ export function getNextCompanyPosition(
 }
 
 export function quitJob(this: IPlayer, company: string, _sing = false): void {
-  if (this.isWorking == true && this.workType !== WorkType.Company && this.companyName == company) {
+  if (this.isWorking === true && this.workType === WorkType.Company && this.companyName === company) {
```

Comment:
Because of the logic in the current version, part-time jobs (`WorkType.CompanyPartTime`) finish when you quit the company but not full-time jobs (`WorkType.Company`). After this change the full-time jobs works but not the part-time ones. We should check that `this.workType` is either part-time or full-time like in [here](https://github.com/danielyxie/bitburner/pull/3874/files#diff-812beaabeb59591d8e0779204c7089eeda40d5ebbc2b80e017ef246285f91a7eR1859).

Reply:
Good catch! Updated.